ui/src/dashboards/containers/DashboardPage.tsx

```diff
@@ -278,20 +281,33 @@ class DashboardPage extends Component<Props, State> {
     return (
       <Page>
         <OverlayTechnology visible={showCellEditorOverlay}>
-          <CellEditorOverlay
-            source={source}
-            sources={sources}
-            notify={notify}
-            fluxLinks={fluxLinks}
-            cell={selectedCell}
-            dashboardID={dashboardID}
-            queryStatus={cellQueryStatus}
-            onSave={this.handleSaveEditedCell}
-            onCancel={this.handleHideCellEditorOverlay}
-            templates={templatesIncludingDashTime}
-            editQueryStatus={this.props.editCellQueryStatus}
-            dashboardTimeRange={timeRange}
-          />
+          <Provider
+            inject={
```

Comment:
How about we store a private `timeMachineContainer` property on the `DashboardPage` instead (of type `TimeMachineContainer`.

Then we can initialize it in the `handleShowCellEditorOverlay` method, e.g.

```
  private handleShowCellEditorOverlay = (cell: DashboardsModels.Cell): void => {
    this.timeMachineContainer = new TimeMachineContainer({
      ...initialStateFromCell(this.state.selectedCell),
      timeRange,
    })

    this.setState({selectedCell: cell, showCellEditorOverlay: true})
  }

```

That way it would only be initialized when it needs to be, and we would avoid having as much logic in the `render` method.

Reply:
Good idea! Much cleaner